pkg/controller/awsmachineclass.go

```diff
@@ -215,44 +217,44 @@ func (c *controller) reconcileClusterAWSMachineClass(class *v1alpha1.AWSMachineC
 		return fmt.Errorf("Retry as machine objects are still referring the machineclass")
 	}
 
-	return c.deleteAWSMachineClassFinalizers(class)
+	return c.deleteAWSMachineClassFinalizers(ctx, class)
 }
 
 /*
 	SECTION
 	Manipulate Finalizers
 */
 
-func (c *controller) addAWSMachineClassFinalizers(class *v1alpha1.AWSMachineClass) error {
+func (c *controller) addAWSMachineClassFinalizers(ctx context.Context, class *v1alpha1.AWSMachineClass) error {
 	clone := class.DeepCopy()
 
 	if finalizers := sets.NewString(clone.Finalizers...); !finalizers.Has(DeleteFinalizerName) {
 		finalizers.Insert(DeleteFinalizerName)
-		return c.updateAWSMachineClassFinalizers(clone, finalizers.List())
+		return c.updateAWSMachineClassFinalizers(ctx, clone, finalizers.List())
 	}
 	return nil
 }
 
-func (c *controller) deleteAWSMachineClassFinalizers(class *v1alpha1.AWSMachineClass) error {
+func (c *controller) deleteAWSMachineClassFinalizers(ctx context.Context, class *v1alpha1.AWSMachineClass) error {
 	clone := class.DeepCopy()
 
 	if finalizers := sets.NewString(clone.Finalizers...); finalizers.Has(DeleteFinalizerName) {
 		finalizers.Delete(DeleteFinalizerName)
-		return c.updateAWSMachineClassFinalizers(clone, finalizers.List())
+		return c.updateAWSMachineClassFinalizers(ctx, clone, finalizers.List())
 	}
 	return nil
 }
 
-func (c *controller) updateAWSMachineClassFinalizers(class *v1alpha1.AWSMachineClass, finalizers []string) error {
+func (c *controller) updateAWSMachineClassFinalizers(ctx context.Context, class *v1alpha1.AWSMachineClass, finalizers []string) error {
 	// Get the latest version of the class so that we can avoid conflicts
-	class, err := c.controlMachineClient.AWSMachineClasses(class.Namespace).Get(class.Name, metav1.GetOptions{})
+	class, err := c.controlMachineClient.AWSMachineClasses(class.Namespace).Get(ctx, class.Name, metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
 
 	clone := class.DeepCopy()
 	clone.Finalizers = finalizers
-	_, err = c.controlMachineClient.AWSMachineClasses(class.Namespace).Update(clone)
+	_, err = c.controlMachineClient.AWSMachineClasses(class.Namespace).Update(context.TODO(), clone, metav1.UpdateOptions{})
```

Comment:
```suggestion
	_, err = c.controlMachineClient.AWSMachineClasses(class.Namespace).Update(ctx, clone, metav1.UpdateOptions{})
```
Check other such files